### PR TITLE
[Core] Mouse key kinetic mode fix

### LINKED ISSUE #13711: [Bug] MOUSEKEY_BASE_SPEED has no effect in Kinetic mode
## Describe the Bug

In Kinetic mode, as per the document, `MOUSEKEY_BASE_SPEED` controls the maximum mouse speed. However, no matter which value I give it, the maximum mouse speed seems to be the same. However, if I increase MOUSEKEY_MOVE_DELTA, the maximum mouse speed increases.

My config:

```
#define MK_KINETIC_SPEED
#define MOUSEKEY_INTERVAL 8
#define MOUSEKEY_MOVE_DELTA 4
#define MOUSEKEY_INITIAL_SPEED 1
#define MOUSEKEY_BASE_SPEED 100
```

My goal is to increase the duration from pressing a mouse cursor button to reach its maximum speed.

Thanks!

## System Information

 - Keyboard:  Moonlander Mark I 
 - Operating system: Arch Linux
 - AVR GCC version: 11.1.0
 - ARM GCC version: 11.1.0
 - QMK Firmware version: 0.13.26
 - Any keyboard related software installed? 
   - [ ] AutoHotKey
   - [ ] Karabiner
   - [ ] Other:

### PR DESCRIPTION
**Pull request by filterpaper**
## Description

Iterating the mouse key kinetic mode fix from #16951 with functional mouse wheel and slightly faster and usable defaults.

## Types of Changes

- [x] Core
- [x] Bugfix
- [x] Documentation

## Issues Fixed or Closed by This PR

* fixes #13711

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Lint is upset.  File needs a format pass.

File: quantum/mousekey.c [hunk 1]
```diff
@@ -66,11 +66,18 @@ uint8_t mk_time_to_max = MOUSEKEY_TIME_TO_MAX;
 /* milliseconds between the initial key press and first repeated motion event (0-2550) */
 uint8_t mk_wheel_delay = MOUSEKEY_WHEEL_DELAY / 10;
 /* milliseconds between repeated motion events (0-255) */
+#    ifdef MK_KINETIC_SPEED
+float   mk_wheel_interval    = 1000.0f / MOUSEKEY_WHEEL_INITIAL_MOVEMENTS;
```
Comment (drashna) [inline quantum/mousekey.c:70]:
```suggestion
float mk_wheel_interval = 1000.0f / MOUSEKEY_WHEEL_INITIAL_MOVEMENTS;
```

File: quantum/mousekey.c [hunk 2]
```diff
@@ -66,11 +66,18 @@ uint8_t mk_time_to_max = MOUSEKEY_TIME_TO_MAX;
...
+#    else
 uint8_t mk_wheel_interval    = MOUSEKEY_WHEEL_INTERVAL;
```
Comment (drashna) [inline quantum/mousekey.c:72]:
```suggestion
uint8_t mk_wheel_interval = MOUSEKEY_WHEEL_INTERVAL;
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by filterpaper** [comment]
> Lint is upset. File needs a format pass.

Can't have nice spacing with linter around.

**Review by drashna — APPROVED** [review]

**Comment by draffensperger** [comment]
I tried this branch out on my Moonlander and it works well for me! I like the kinetic mouse keys algorithm - helps with both slow and fast movements better. Any chance this could get final review and merged in?

**Comment by filterpaper** [comment]
This PR is will be merged into `develop` branch. Barring bugs, it should be in `master` by August 27, 2022.

**Review by fauxpark — APPROVED** [review]

**Pull request merged**